src/features/defi/components/PairIcons/PairIcons.tsx

```diff
@@ -12,19 +15,43 @@ export const PairIcons = ({
   iconBoxSize?: string
   iconSize?: ThemingProps<'Avatar'>['size']
 } & FlexProps): JSX.Element => {
-  const bg = useColorModeValue('gray.200', 'gray.700')
+  const firstIcon = icons[0]
+  const remainingIcons = useMemo(() => {
+    const iconsMinusFirst = icons.slice(1)
+    if (iconsMinusFirst.length > 1) {
+      return (
+        <Center position='relative' overflow='hidden' borderRadius='full' ml='-2.5'>
+          {iconsMinusFirst.map((iconSrc, index) => (
+            <Avatar
+              key={iconSrc}
+              src={iconSrc}
+              position='absolute'
+              left={index * -3}
+              top={index * -3}
+              filter='blur(10px)'
+            />
+          ))}
+          <Avatar
+            bg='whiteAlpha.100'
+            borderRadius='none'
+            color='text.base'
+            textShadow='sm'
+            name={`+ ${iconsMinusFirst.length}`}
+            boxSize={iconBoxSize}
+            size={iconSize}
+            sx={assetIconSx}
+          />
+        </Center>
+      )
+    }
+    return iconsMinusFirst.map(iconSrc => (
+      <AssetIcon ml='-2.5' key={iconSrc} src={iconSrc} boxSize={iconBoxSize} size={iconSize} />
```

Comment:
Q: do we want to update `<AssetIcon />` to *not* support multiple icons as a follow-up?

Reply:
Not relevant anymore